src/models/Location.java

```diff
@@ -6,97 +6,278 @@
 import java.util.stream.Collectors;
 
 public class Location {
-    private final String name;
-
-    private int x, y;
-    private Guard invariant;
-    private CDD inconsistentPart;
-
-    // Must be final as Automaton expects it to be constant through the lifetime
-    private final boolean isInitial;
-    private boolean isUrgent;
-    private boolean isUniversal;
-    private boolean isInconsistent;
-
-    public Location(String name, Guard invariant, boolean isInitial, boolean isUrgent, boolean isUniversal, boolean isInconsistent, int x, int y) {
+    protected String name;
+    protected int x, y;
+
+    protected Guard invariantGuard;
+    protected CDD invariantCdd;
```

Comment:
Both are required because we want to 1. allow the creation of locations without initializing the CDD lib and 2. storing them is used to reduce the conversion between the two types.

Reply:
Usefull information to be put in javadoc. I haven't checked the actual changes yet.